ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -170,19 +177,24 @@ private class SqlLedger(
               ledgerDao
                 .storeLedgerEntry(offset, offset + 1, ledgerEntryGen(offset))
                 .map(_ => ())(DEC)
+                .recover {
+                  case t =>
+                    //recovering from the failure so the persistence stream doesn't die
+                    logger.error(s"Failed to persist entry with offset: $offset", t)
+                    ()
```

Comment:
That's merely ignoring the error, right? Which means that we actually lose the ledger entry. We probably should rather retry n times and then crash the application?

Reply:
I am not sure. Why can't the command just fail, and we put the retry responsibility on the client? I tend to shy away from ad-hoc retry logics, because you have to make arbitrary decisions which you might not be entitled to. In this case the commands in-flight have time related constraints for instance, so how long should we retry them?

Reply:
Another thing is that a Postgres database can be down for a long time. A robust application should survive that and recover when it comes back.

Reply:
Is the command/transaction actually rejected? This would then trigger another write to the database that would fail, right?

Reply:
Discussed offline: the postgres transaction will fail and thus be rolled back by `HikariJdbcConnectionProvider`. Since the client will never get a completion or a transaction, it should run into a command timeout and retry the command.